darts/timeseries.py

```diff
@@ -140,156 +582,251 @@ def pd_dataframe(self, copy=True) -> pd.DataFrame:
         Returns
         -------
         pandas.DataFrame
-            The Pandas Dataframe underlying this time series
+            The Pandas DataFrame representation of this time series
         """
+        self._assert_deterministic()
         if copy:
-            return self._df.copy()
+            return pd.DataFrame(self._xa[:, :, 0].values.copy(),
```

Comment:
xarray has to_pandas() functionality. Wouldn't that work here?

Reply:
Oh yes maybe, but at least here we are really controlling how the export is done, and not relying on some external xarray logic which might perhaps do some hidden things we don't know about.